src/main/java/io/vertx/core/http/impl/Http2ServerConnection.java

```diff
@@ -170,15 +162,16 @@ VertxHttp2Stream<?> stream(int id) {
   protected synchronized void onHeadersRead(int streamId, Http2Headers headers, StreamPriority streamPriority, boolean endOfStream) {
     VertxHttp2Stream stream = stream(streamId);
     if (stream == null) {
-      if (isMalformedRequest(headers)) {
+      HostAndPort authority = isMalformedRequest(headers);
```

Comment:
This is still not preventing the authority to be validated again, but just making a lighter approach on validation: the ideal would be to check if the new stream matches with an existing last received valid authority and skip it.
I have verified that there are no hooks to detect if the authority is received fresh new or just cached by HPACK, so it means that we should do it ourself..